src/cmds/scripts/pbs_postinstall.in

```diff
@@ -319,7 +319,7 @@ install_pbsinitd() {
 		else
 			initscript="/etc/rc.d/init.d/pbs"
 		fi
-		cp ${PBS_EXEC}/libexec/pbs_init.d $initscript
+		ln -s ${PBS_EXEC}/libexec/pbs_init.d $initscript
```

Comment:
If an admin edits the init script they will modify the one in PBS_EXEC/libexec with this change in place. If they break it, they can't simply copy the original version to fix the problem. A brief look at the LSB 3.0 document indicates that libexec is intended for executables that are not intended to be invoked by users. It doesn't explicitly state these files should not be modified, but I think that is implied. Files under /etc, on the other hand, are allowed to be modified.

It comes down to interpretation of the standards as to whether using a symbolic link is the correct thing to do here. I'm not entirely opposed to it, but I think others need to agree that this change is appropriate. I tend to believe it is not.

Reply:
The problem we have today is two different ways of starting/stopping pbs. The admin can do it using the legacy pbs init script resided in `/etc/init.d` or using systemctl. But the systemctl refers to the copy of the file present in libexec directory.
If someone wants to make some site-specific changes to init script, they need to make it in two places. Otherwise one of them will be outdated. The simplest solution for this problem without breaking the backward compatibility is to create a symlink to the original file, rather than a copy.

It's the responsibility of the person who is editing the file to create a copy of it so that they can use it in case of failure. I don't think PBSPro should leave a copy behind as a backup which one can always extract from the rpms.

Reply:
I have never seen a symbolic link in /etc/init.d for any service. Of course there are symbolic links in directories like /etc/rc3.d that point to the files in /etc/init.d. I imagine these SysV init scripts will go the way of the dinosaur at some point, so it probably doesn't matter all that much. I still don't agree with the approach, but not enough to request changes or block merging. Other reviewers are free to formulate their own opinions. Thanks for the explanation.

Reply:
I'm personally ok with this change, but i guess this is really a question for the admins and if they'll be okay with it. Speaking of which, this might be an interface change and need documenting, shouldn't we be posting this on the forum?

Reply:
I've created a [thread ](community.pbspro.org/t/creating-symlink-to-etc-init-d-pbs-instead-of-a-separate-copy/1319
)for this discussion.

Reply:
I just replied to the forum. I think we should resolve this by cleaning up the boot check file. No need to change the start/stop script to a symbolic link at this time. It seems to be working fine as it is.